cli/config_file.go

```diff
@@ -0,0 +1,292 @@
+package main
+
+import (
+	"crypto/rand"
+	"encoding/base64"
+	"errors"
+	"fmt"
+	"log"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"gopkg.in/yaml.v3"
+)
+
+// Settings and state of the application
+type Settings struct {
+	SettingsFile     SettingsFile
+	SettingsWrite    bool
+	SettingsExists   bool
+	SettingsFilePath string
+	Command          string
+	NetworkGatewayIP string
+	Containers
+}
+
+type Containers struct {
+	Crux           ContainerSettings
+	CruxMigrate    ContainerSettings
+	CruxUI         ContainerSettings
+	Kratos         ContainerSettings
+	KratosMigrate  ContainerSettings
+	CruxPostgres   ContainerSettings
+	KratosPostgres ContainerSettings
+	MailSlurper    ContainerSettings
+}
+
+type ContainerSettings struct {
+	Image    string
+	Name     string
+	Disabled bool
+	CruxAddr string
+}
+
+// Settings file will be read/written as this struct
+type SettingsFile struct {
+	// version as in image tag like "latest" or "stable"
+	Version        string `yaml:"version"`
+	CruxDisabled   bool   `yaml:"crux_disabled"`
+	CruxUIDisabled bool   `yaml:"crux-ui_disabled"`
+	Network        string `yaml:"network-name"`
+	Prefix         string `yaml:"prefix"`
+	Options
+}
+
+type Options struct {
+	TimeZone               string `yaml:"timezone"`
+	CruxAgentGrpcPort      uint   `yaml:"crux-agentgrpc-port"`
+	CruxGrpcPort           uint   `yaml:"crux-grpc-port"`
+	CruxUIPort             uint   `yaml:"crux-ui-port"`
+	CruxSecret             string `yaml:"crux-secret"`
+	CruxPostgresPort       uint   `yaml:"cruxPostgresPort"`
+	CruxPostgresDB         string `yaml:"cruxPostgresDB"`
+	CruxPostgresUser       string `yaml:"cruxPostgresUser"`
+	CruxPostgresPassword   string `yaml:"cruxPostgresPassword"`
+	KratosAdminPort        uint   `yaml:"kratosAdminPort"`
+	KratosPublicPort       uint   `yaml:"kratosPublicPort"`
+	KratosPostgresPort     uint   `yaml:"kratosPostgresPort"`
+	KratosPostgresDB       string `yaml:"kratosPostgresDB"`
+	KratosPostgresUser     string `yaml:"kratosPostgresUser"`
+	KratosPostgresPassword string `yaml:"kratosPostgresPassword"`
+	KratosSecret           string `yaml:"kratosSecret"`
+	MailSlurperPort        uint   `yaml:"mailSlurperPort"`
+	MailSlurperPort2       uint   `yaml:"mailSlurperPort2"`
+}
+
+const DefaultPostgresPort = 5432
+
+const DefaultCruxAgentGrpcPort = 5000
+const DefaultCruxGrpcPort = 5001
+const DefaultCruxUIPort = 3000
+const DefaultCruxPostgresPort = 5432
+const DefaultKratosPublicPort = 4433
+const DefaultKratosAdminPort = 4434
+const DefaultKratosPostgresPort = 5433
+const DefaultMailSlurperPort = 4436
+const DefaultMailSlurperPort2 = 4437
+
+const SecretLength = 32
+
+const bufferMultiplier = 2
+
+const FilePerms = 0600
+const DirPerms = 0750
+
+const SettingsFileName = "settings.yaml"
+const SettingsFileDir = "dyo"
```

Comment:
I don't know which one would be better but prefer something like 'dyo-cli' or 'dyrector.io'

Reply:
'dyo-cli' would be fine I think 🤔

Reply:
"dyo" as default settings directory was changed to dyo-cli.